src/QuizEnd.tsx

```diff
@@ -1,36 +1,37 @@
-import { useNavigate } from "react-router-dom";
-import { useQuiz } from "./QuizProvider";
-import { Header } from "./Header";
-import { Button, Heading } from "@chakra-ui/react";
-export function QuizEnd() {
-  const {
-    state,
-    dispatch,
-    setUserClick,
-    setShowNextBtn,
-    setSelected
-  } = useQuiz();
-  const navigate = useNavigate();
-  return (
-    <>
-      <Header />
-      <Heading as="h3" size="lg">
-        Questions Correct : {state.score / 10}/10
-      </Heading>
-      <Heading as="h4" size="lg">
-        Total score :{state.score} /100
-      </Heading>
-      <Button
-        onClick={() => {
-          dispatch({ type: "RESET" });
-          navigate("/");
-          setUserClick(false);
-          setShowNextBtn(false);
-          setSelected("");
-        }}
-      >
-        Play Again
-      </Button>
-    </>
-  );
-}
+  import { useNavigate } from "react-router-dom";
+  import { useQuiz } from "./QuizProvider";
+  import { Header } from "./Header";
+  import { Button, Heading } from "@chakra-ui/react";
+
+
+  export function QuizEnd() {
+    const {
+      state,
+      dispatch,
+      setUserClick,
+      setShowNextBtn,
+      setSelected
+    } = useQuiz();
+    const navigate = useNavigate();
+
+    return (
+      <>
+        <Header />
+        <Heading as="h3" size="lg">
+          Questions Correct : {state.score / 10}/10
+        </Heading>
+        <Heading as="h4" size="lg">
+          Total score :{state.score} /100
+        </Heading>
+        <Button onClick={() => {
```

Comment:
for better code structure  what you can do is create a function above, put these operations in there and then call that on onClick() here